### ISSUE TITLE
gem conflict with ChefDK 2+

### ISSUE THREAD

**Issue by melhossarySG** [test-kitchen/kitchen-google#51]
Hello, 

I am not able to get the kitchen to work at all and it is driving me nuts

If i try to uninstall the google-api-client-0.8.7 the whole chefdk fails and stop working and I have to re-install it, 

```
[2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: ------Exception-------
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: Class: Kitchen::ClientError
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: Message: Could not load the 'gce' driver from the load path. Please ensure that your driver is installed as a gem or included in your Gemfile if using Bundler.
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: ----------------------
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: ------Backtrace-------
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/driver.rb:48:in `rescue in for_plugin'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/driver.rb:38:in `for_plugin'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:233:in `new_driver'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:246:in `new_instance'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:142:in `block in build_instances'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `map'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `with_index'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `build_instances'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:117:in `instances'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command.rb:112:in `filtered_instances'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command.rb:142:in `parse_subcommand'
E, [2017-07-20T11:22:52.053570 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command/list.rb:30:in `call'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:53:in `perform'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:121:in `list'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:334:in `invoke_task'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/bin/kitchen:13:in `block in <top (required)>'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/errors.rb:171:in `with_friendly_errors'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/bin/kitchen:13:in `<top (required)>'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/bin/kitchen:275:in `load'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/bin/kitchen:275:in `<main>'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: ----End Backtrace-----
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: ---Nested Exception---
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: Class: Gem::ConflictError
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: Message: Unable to activate kitchen-google-1.2.0, because google-api-client-0.8.7 conflicts with google-api-client (~> 0.9.0)
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: ----------------------
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: ------Backtrace-------
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/driver.rb:48:in `rescue in for_plugin'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/driver.rb:38:in `for_plugin'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:233:in `new_driver'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:246:in `new_instance'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:142:in `block in build_instances'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `map'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `with_index'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:141:in `build_instances'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/config.rb:117:in `instances'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command.rb:112:in `filtered_instances'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command.rb:142:in `parse_subcommand'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/command/list.rb:30:in `call'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:53:in `perform'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:121:in `list'
E, [2017-07-20T11:22:52.054573 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/cli.rb:334:in `invoke_task'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/bin/kitchen:13:in `block in <top (required)>'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/lib/kitchen/errors.rb:171:in `with_friendly_errors'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.16.0/bin/kitchen:13:in `<top (required)>'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/bin/kitchen:275:in `load'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: C:/opscode/chefdk/bin/kitchen:275:in `<main>'
E, [2017-07-20T11:22:52.055575 #2332] ERROR -- Kitchen: ----End Backtrace-----


If i try to uninstall the google-api-client-0.8.7 the whole chefdk fails and stop working and I have to re-install it, 

C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'google-api-client' (~> 0.8.0) - did find: [google-api-client-0.13.1] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=C:/Users/mohamed.elhossary/AppData/Local/chefdk/gem/ruby/2.4.0;C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0', execute `gem env` for more information
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/specification.rb:1442:in `block in activate_dependencies'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/specification.rb:1431:in `each'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/specification.rb:1431:in `activate_dependencies'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/specification.rb:1413:in `activate'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
        from C:/opscode/chefdk/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
        from C:/opscode/chefdk/bin/chef:40:in `<main>'

```

Could you please help ??

**Comment by btm** [comment]
https://github.com/chef/chef-provisioning-fog depends on https://github.com/fog/fog which depends on https://github.com/fog/fog-google which is currently pinned to `google-api-client ~> 0.8.7`.

Updating that pin is being worked on by the Fog community in https://github.com/fog/fog-google/issues/226. Once that is complete we can update to whatever version of fog consumes that in https://github.com/chef/chef-provisioning-fog and ChefDK should end up with a newer version of https://github.com/google/google-api-ruby-client.

**Comment by gionn** [comment]
upstream fix have been released as fog-google 1.0.0

**Comment by tas50** [comment]
We fixed this up, but it's going to be a constant battle unless google goes to 1.0 on the api.
